localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -207,26 +207,33 @@ void EKFLocalizer::timerCallback()
 
   const double x = ekf_.getXelement(IDX::X);
   const double y = ekf_.getXelement(IDX::Y);
-  const double z = z_filter_.get_x();
+
+  const double current_pitch = pitch_filter_.get_x();
+  const double new_pitch = current_pitch + pitch_rate_ * dt * cntTimerCallback_;
+  cntTimerCallback_++;
 
   const double biased_yaw = ekf_.getXelement(IDX::YAW);
   const double yaw_bias = ekf_.getXelement(IDX::YAWB);
 
   const double roll = roll_filter_.get_x();
-  const double pitch = pitch_filter_.get_x();
   const double yaw = biased_yaw + yaw_bias;
   const double vx = ekf_.getXelement(IDX::VX);
   const double wz = ekf_.getXelement(IDX::WZ);
 
+  const double z = z_filter_.get_x();
+  const double val_sin = -std::sin(pitch_from_ndt_);
+  const double z_addition = val_sin * vx * dt * cntTimerCallback_;
+  const double new_z = z + z_addition;
```

Comment:
It is a bit difficult to catch up what is going on in this field. (Not only your fault, but mainly due to the current EKF's implementation, sorry for that :pray: )

This is just an idea, but what about this?
```cpp
  /* pose measurement update */
  if (!pose_queue_.empty()) {
    ...
  }

  /* twist measurement update */
  if (!twist_queue_.empty()) {
    ...
  }

  /* z and pitch update for slopes */
  // UPDATE Z AND PITCH HERE

  /* Extract the results */
  const double x = ekf_.getXelement(IDX::X);
  ... (rest of the code)
```

Reply:
It's a good idea.
I will implement your advice.